### ISSUE TITLE
crossOrigin attribute being disabled in html5 engine

### ISSUE THREAD

**Issue by danrossi** [flowplayer/flowplayer#993]
Hi I am not sure what that feature is for but the html5 engine is disabling any reference to the crossOrigin attributed required for CORS features on the mediaelement. 

https://github.com/flowplayer/flowplayer/blob/master/lib/engine/html5.js#L130

I currently have to do this workaround to stop it doing this. 

```
video.setAttribute("crossorigin","anonymous");

            player.on("load", function(e) {

                hasNativeSubtitles = !!player.conf.nativesubtitles;
                hasSubtitles = !!flowplayer.support.subtitles;
                player.conf.nativesubtitles = true;
                flowplayer.support.subtitles = false;

            }).on("ready", function(e) {
                player.conf.nativesubtitles = hasNativeSubtitles;
                flowplayer.support.subtitles = hasSubtitles;
            });
```

Thoughts ? 

Obviously not of concern for Safari/Webkit as there is a ticket just for this and there hasn't been any movement on that for years.

**Comment by nnarhinen** [comment]
When using native subtitles, flowplayer will set the crossorigin attribute if needed. Link to a setup that does not work and we will investigate.

The attribute is removed, because when it is used when no cross origin subtitles are loaded, it prevents the whole video from being shown. IIRC in firefox, but anyway the behavior was not consistent between browsers.

**Comment by danrossi** [comment]
I'm sorry I just saw this and it's gone..

I have no idea what is going on with subtitles sorry.

As far as Firefox goes it still has not got a cuechange event anyway, when I was dealing with texttrack features Firefox had a bug ticket to implement that so I reverted back to player cuepoints for it. For everything else I am loading vtt externally, parsing and adding using the texttrack api because of CORS inconsistencies. 

CORS support for texttracks from memory is still broken in Safari and IE. Webkit are still meddling with CORS support fullstop for video after years of inaction. I believe crossorigin would go on the texttrack for that rather than on the video ? 

It is needed for canvas drawing and WebGL for anything other than Safari. So manipulated whatever the video is displaying, a basic requirement. 

I just found another change now broke another work around but I manage to work around the work around for now hah !

Hopefully there is enough info, I don't believe I had to spend time creating a fiddle but I am find one I have done for Android webgl/canvas bugs.  It would work the same.
